DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -304,13 +304,22 @@ public function applyDefaults(ContainerBuilder $container, $serviceId, array $at
 
         $definition->addMethodCall('setLabel', array($label));
 
+
+        $persistFilters = $container->getParameter('sonata.admin.configuration.filters.persist');
+        // override default configuration with admin config if set
         if (isset($attributes['persist_filters'])) {
-            $persistFilters = (bool) $attributes['persist_filters'];
-        } else {
-            $persistFilters = (bool) $container->getParameter('sonata.admin.configuration.filters.persist');
+            $persistFilters = $attributes['persist_filters'];
         }
 
-        $definition->addMethodCall('setPersistFilters', array($persistFilters));
+        // if boolean config value provided, use the default persister : session
+        if (is_bool($persistFilters) && $persistFilters) {
```

Comment:
Do we have to check if `is_bool($persistFilters)`. Can't we instead just check normally? This property is set from container parameter or from the attributes, and we can cast it previously on the other case.

Maybe we can change from scalarNode to booleanNode on the Bundle Configuration?

Just to have properties that we know they are a concrete type and can't be other things.